test/color_test.py

```diff
@@ -990,6 +1006,7 @@ def test_pickle(self):
     # only available if ctypes module is also available
 
     @unittest.skipIf(IS_PYPY, "PyPy has no ctypes")
+    @_suppress_warnings(DeprecationWarning)
     def test_arraystruct(self):
```

Comment:
Same here. ArrayInterface(color[:i]) will work perhaps. But this is not a test of set_length functionality, it's an incidental use, so it should be removed.

Reply:
Apparently ``np.array(color[:3])`` also works...

Reply:
@Starbuck5 I couldn't make it work (your proposition doesn't work, only after some modifications I managed to make it working), but after that I discovered that a lot of sub-checks in that part failed. I started fixing by making modifications for that part of code, but I think there is no need to change anything, as the ``test_arraystruct`` performs (I think) some checks with that ``color.set_length``. Probably there is something that I can do there, but I don't want to touch that part of the code, as I don't fully understand it...

Reply:
This is how I understand this test:
It checks what structure format Color object has with different color lengths, so changing the Color object to tuple and test it, is pointless because: tuple has different arrangement and checking the structure of tuple provides not a very valuable information. So I think this test is solely for testing ``set_length`` functionality and should be removed in the future.